gateway-js/src/compose.ts

```diff
@@ -0,0 +1,40 @@
+import { resolve } from 'path';
+import { Worker } from 'worker_threads';
+import { buildSchema, GraphQLError, print } from 'graphql';
+import type { composeAndValidate, ComposedGraphQLSchema, ServiceDefinition } from '@apollo/federation';
+import type { WorkerCompositionResult } from './compose-worker';
```

Comment:
Perhaps we should gate our usage of `worker_threads` on its presence?  I know you're aware that `worker_threads` is a Node.js-ism, but many of our users run the gateway in non-Node.js environments.  Providing a polyfill for `worker_threads` that works in a barebones JS runtime (e.g., V8, whatever variation of V8 Cloudflare Workers runs) is an interesting conversation (plausibly polyfilled with a happy-degradation path when not present, I think?), but adds the burden of that configuration to consumers.  (This is usually amplified by naïve/incomplete configurations that don't compensate for modules other than the most popular.)  Not to say that those configurations shouldn't be fixed, but we often face the need to advise/assist in those cases, as was the case with our introduction of WASM and the resulting https://github.com/apollographql/federation/issues/255.

**My suggestion is merely** to utilize a conditional `import` based on the presence of the `worker_threads` module (e.g. `if (require.resolve('worker_threads'))`?  While conditional/nested-`import`s don't work out of the box JavaScript, TypeScript should transpile to the appropriate `require`s.